addons/mail/models/res_partner.py

```diff
@@ -126,16 +126,25 @@ def get_static_mention_suggestions(self):
         return [partner.mail_partner_format() for partner in partners]
 
     @api.model
-    def get_mention_suggestions(self, search, limit=8, channel_id=None):
+    def get_mention_suggestions(self, search, limit=8):
         """ Return 'limit'-first partners' such that the name or email matches a 'search' string.
             Prioritize partners that are also users, and then extend the research to all partners.
-            If channel_id is given, only members of this channel are returned.
             The return format is a list of partner data (as per returned by `mail_partner_format()`).
         """
+        partners = self._get_mention_suggestions(search, limit)
+        return [partner.mail_partner_format() for partner in partners]
+
+    @api.model
+    def get_channel_mention_suggestions(self, search, limit=8, channel_id=None):
```

Comment:
`channel_id` should be required on this case, it should even be the first param I think